### PR TITLE
Added option to Percentage price scale

### PR DESCRIPTION
**Pull request by Rav3n95**

### CODE REVIEW

**Comment by GhzGarage** [comment]
I'll leave this to you @ItsANoBrainer so you can decide. it has conflicts anyway

File: client/cl_ui.lua [hunk 1]
```diff
@@ -287,14 +287,24 @@ function InitiateMenus(isMotorcycle, vehicleHealth, categories, welcomeLabel)
                     tempNum = tempNum + 1
 
                     if maxVehiclePerformanceUpgrades == 0 then
-                        populateMenu(v.category:gsub("%s+", "") .. "Menu", n.id, n.name, "$" .. vehicleCustomisationPrices.performance.prices[tempNum])
+
+                        if Config.UsePercentage then
+                            populateMenu(v.category:gsub("%s+", "") .. "Menu", n.id, n.name, "$" .. math.floor((vehiclePrice / 100) * vehicleCustomisationPricesPercentage.performance[tempNum]))
+                        else
+                            populateMenu(v.category:gsub("%s+", "") .. "Menu", n.id, n.name, "$" .. vehicleCustomisationPrices.performance.prices[tempNum])
+                        end
```
Comment (ItsANoBrainer) [inline client/cl_ui.lua:295]:
```suggestion
local price = Config.UsePercentage and math.floor((vehiclePrice / 100) * vehicleCustomisationPricesPercentage.performance[tempNum]) or vehicleCustomisationPrices.performance.prices[tempNum]
populateMenu(v.category:gsub("%s+", "") .. "Menu", n.id, n.name, "$" ..price )
```
If you could make this change to all the entries to make it "cleaner" looking.

**Review by ItsANoBrainer — CHANGES_REQUESTED** [review]
Left an example of a cleaner version of doing the Config price. If you can change all the entries to a version like that, and also remove the default changes to shared/locations.lua, I approve! Also need conflicts fixed.

File: config.lua [hunk 2]
```diff
@@ -28,6 +30,23 @@ vehicleCustomisationPrices = {
     plateindex = {price = 1000}
 }
 
+vehicleCustomisationPricesPercentage = {
+    cosmetics = 5,
+    respray = 2,
+    performance = {0, 3, 6, 9, 12, 15},
```
Comment (ItsANoBrainer) [inline config.lua:36]:
```suggestion
    performance = {0, 3, 6, 9, 12, 15, 20},
```

**Review by ItsANoBrainer — CHANGES_REQUESTED** [review]

**Comment by Rav3n95** [comment]
Sorry i been a bit busy whit qb-garages polyzone rework. Soonest i done whit that, i be ready to rework this to new version. Also thanks a lot for helping out.

**Comment by Rav3n95** [comment]
Updated, tested only alone.

File: client/cl_bennys.lua [hunk 3]
```diff
@@ -32,12 +32,26 @@ local originalTurboState = nil
...
 
 -----------------------
 ----   Functions   ----
 -----------------------
 
 --#[Local Functions]#--
+local function GetVehiclePrice()
+    local vehicle = GetVehiclePedIsIn(PlayerPedId(), false)
+
+    for k, v in pairs(QBCore.Shared.Vehicles) do
```
Comment (uShifty) [inline client/cl_bennys.lua:45]: Replace tis loop with this 
`local hash = GetEntityModel(vehicle)`
`vehiclePrice = QBShared.VehicleHashes[hash] and QBShared.VehicleHashes[hash].price or vehicleBasePrice`

**Review by uShifty — CHANGES_REQUESTED** [review]

File: client/cl_bennys.lua (hunk 3 above)
Comment (Rav3n95) [reply client/cl_bennys.lua]: Thanks

**Comment by Z3rio** [comment]
Not needed PR considering we are going to rewrite qb-customs anyways.

**Pull request closed without merging**